native/src/main/java/io/ballerina/stdlib/http/api/service/signature/QueryParam.java

```diff
@@ -74,4 +80,15 @@ public int getEffectiveTypeTag() {
     public boolean isArray() {
```

Comment:
Cant we have a parent type for these `QueryParam`, `PathParam`, `HeaderParam`. Saw some duplicate features like `isArray`, `token`?

Reply:
Yes, that would be better. Will address this 👍🏼

Reply:
Fixed via [b2aa353](https://github.com/ballerina-platform/module-ballerina-http/pull/1608/commits/b2aa353bb917729ab479676d8b1a64081ccd94f0)